test/blackbox-tests/test-cases/pkg/outdated.t

```diff
@@ -0,0 +1,53 @@
+  $ . ./helpers.sh
+
+  $ outdated () {
+  >  dune pkg outdated --opam-repository-path=mock-opam-repository $@
+  > }
+
+`dune pkg outdated` lists the outdated packages in the current project.
+  $ mkrepo
+  $ mkpkg foo
+  $ mkpkg bar <<EOF
+  > depends: [ "foo" {>= "0.0.1"} ]
+  > EOF
+  $ solve_project <<EOF
+  > (lang dune 3.11)
+  > (package
+  >  (name baz)
+  >  (depends bar))
+  > EOF
+  Solution for dune.lock:
+  bar.0.0.1
+  foo.0.0.1
+  
+No package should be outdated after a fresh lock.
+  $ outdated
+
+Adding a new version of the bar package to the repository.
+  $ mkpkg bar 0.0.2
+
+Dune should report the new version of bar as available.
+  $ outdated
+  bar 0.0.1 < 0.0.2
+
+Now we add a new version of the foo package to the repository.
+Dune should only report the bar package as it is an immediate dependency.
+  $ mkpkg foo 0.0.2 
+  $ outdated
+  bar 0.0.1 < 0.0.2
+
+If --transitive is also passed then both should be reported.
+  $ outdated --transitive
+  bar 0.0.1 < 0.0.2
+  foo 0.0.1 < 0.0.2
+
+If we remove packages from the repository then we should get a nice error.
+  $ rm -rf mock-opam-repository/packages/bar
+  $ outdated 
+  Error: bar is not available in any of the configured repositories
+  [1]
+
+When printing both successes and failures, the errors should appear afterwards.
+  $ outdated --transitive
+  Error: bar is not available in any of the configured repositories
```

Comment:
I changed some things around and I missed that this has now swallowed its output. Will fix.